src/plugins/vis_type_table/public/components/table_vis_cell.tsx

```diff
@@ -23,6 +23,11 @@ export const getTableVisCellValue = (
   const rawContent = row[columnId];
   const colIndex = columns.findIndex((col) => col.id === columnId);
   const htmlContent = columns[colIndex].formatter.convert(rawContent, 'html');
+  dompurify.addHook('uponSanitizeAttribute', function(currentNode, hookEvent) {
+	if (hookEvent.attrName == 'target'){
+		hookEvent.forceKeepAttr = true;
+	}
+  });
```

Comment:
I don't think we should blindly allow any `target` and we shouldn't allow `target="_blank" without the `rel`.
```suggestion
  dompurify.addHook('uponSanitizeAttribute', (node) => {
    if ('target' in node) {
      node.setAttribute('target', '_blank');
      node.setAttribute('rel', 'noopener noreferrer');
    }
  });
```

Reply:
How can you assume that target should be blank every time. It can be self also. Isn't it ? I can do a check to see the value of the attr and set the rel only if the value is _blank

Reply:
Also even in the code suggested by you, we are looking at all all the nodes with target attr.

Reply:
I have the made the changes related to what I mentioned in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8205#discussion_r1790754687

Reply:
added multiple test cases as well.

Reply:
After further digging, I found that the `target` attribute is managed by the specific formatters. While OSD internally only has the URL formatter that generates anchor tags, other plugins or third-party forks could potentially introduce additional formatters. Therefore, it is the responsibility of each formatter to ensure that a valid `target` value is assigned. However, it is also OSD's responsibility to exercise due diligence and add `rel="noopener noreferrer"` to all links, regardless of whether the `target` attribute is set or its assigned value. This approach will help mitigate potential security risks associated with the improper handling of links.

```
  dompurify.addHook('afterSanitizeElements', (node) => {
    if (node.tagName?.toUpperCase?.() === 'A') {
      node.setAttribute('rel', 'noopener noreferrer');
    }
  });
```

I haven't tested this but lemme know what you think.

Reply:
Why do you want to do this in a new event _afterSanitizeElements_. I would do this as part of _uponSanitizeAttribute_ event. We can add these two values for all the anchor tags when processing the latter event. I agree that adding them to all the anchor tags does help improve the security.

Reply:
@AMoo-Miki  I have made the changes. Can it be merged now ?

Reply:
@AMoo-Miki gentle reminder to get this reviewed.